service/src/main/java/org/cbioportal/service/impl/GenePanelServiceImpl.java

```diff
@@ -172,56 +189,40 @@ public GenePanel getGenePanel(String genePanelId) throws GenePanelNotFoundExcept
                         datum -> true));
 
-        Stream<GenePanelData> genePanelDataStream = fetchGenePanelDataByMolecularProfileIds(molecularProfileIds)
+        return fetchGenePanelDataByMolecularProfileIds(molecularProfileIds)
             .stream()
-            .filter(datum -> queriedMolecularProfileCaseIdentifierMap.containsKey(keyGenerator.apply(datum)));
+            .filter(datum -> queriedMolecularProfileCaseIdentifierMap.containsKey(keyGenerator.apply(datum)))
+            .collect(toList());
 
-        return annotateDataFromSequencedSampleLists(genePanelDataStream, molecularProfileIds);
     }
 
-    private List<GenePanelData> annotateDataFromSequencedSampleLists(Stream<GenePanelData> genePanelData,
-                                                                     Set<String> molecularProfileIds) {
-
-        Map<String, MolecularProfile> sequencedMolecularProfileMap = molecularProfileRepository
-            .getMolecularProfiles(new ArrayList<>(molecularProfileIds), "SUMMARY")
-            .stream()
-            .filter(profile ->
-                profile.getMolecularAlterationType().equals(MolecularProfile.MolecularAlterationType.MUTATION_EXTENDED) ||
-                    profile.getMolecularAlterationType().equals(MolecularProfile.MolecularAlterationType.FUSION) ||
-                    (profile.getMolecularAlterationType().equals(MolecularProfile.MolecularAlterationType.STRUCTURAL_VARIANT) &&
-                        profile.getDatatype().equalsIgnoreCase("FUSION")))
-            .collect(Collectors.toMap(MolecularProfile::getStableId, Function.identity()));
-
-        Map<String, Boolean> sampleSequencedBySampleList = new HashMap<>();
-
-        sequencedMolecularProfileMap
-            .values()
-            .stream()
-            .map(profile -> profile.getCancerStudyIdentifier() + SEQUENCED_LIST_SUFFIX)
-            .distinct()
-            .forEach(sampleListId -> {
-                try {
-                    SampleList sampleList = sampleListService.getSampleList(sampleListId);
-                    sampleList
-                        .getSampleIds()
-                        .forEach(sampleId ->
-                            sampleSequencedBySampleList.put(sampleList.getCancerStudyIdentifier() + sampleId, true));
-                } catch (SampleListNotFoundException ignored) {
-                }
-            });
-
-        if (!sampleSequencedBySampleList.isEmpty()) {
-            return genePanelData
-                .map(datum -> {
-                    if (!datum.getProfiled() && sequencedMolecularProfileMap.containsKey(datum.getMolecularProfileId())) {
-                        datum.setProfiled(sampleSequencedBySampleList.getOrDefault(datum.getStudyId() + datum.getSampleId(), false));
-                    }
-                    return datum;
-                })
-                .collect(toList());
+    private List<GenePanelData> annotateDataFromSequencedSampleLists(List<GenePanelData> genePanelData,
```

Comment:
should be more specific about the annotation?  we are really setting the profiled flag, right?

Reply:
yes

Reply:
Added description for the function